async-request-reply/src/CloudBlockBlobExtensions.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Contoso
+{
+    public static class CloudBlockBlobExtensions
+    {
+        public static string GenerateSASURI(this CloudBlockBlob blob)
+        {
+            SharedAccessBlobPolicy sasConstraints = new SharedAccessBlobPolicy();
+            sasConstraints.SharedAccessStartTime = DateTimeOffset.UtcNow.AddMinutes(-5);
+            sasConstraints.SharedAccessExpiryTime = DateTimeOffset.UtcNow.AddMinutes(10);
+            sasConstraints.Permissions = SharedAccessBlobPermissions.Read;
+
+            //Generate the shared access signature on the blob, setting the constraints directly on the signature.
+            string sasBlobToken = blob.GetSharedAccessSignature(sasConstraints);
+
+            //Return the URI string for the container, including the SAS token.
+            return blob.Uri + sasBlobToken;
```

Comment:
Can you access the Uri property for a Cloud Block Blob that hasn't been created yet?

Reply:
You can but returning the SAS in this function is not required for the pattern. Removed.